var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -213,11 +219,12 @@ def libs(self):
         # to the libraries needed
         query2libraries = {
             tuple(): ['libhdf5'],
-            ('cxx', 'fortran', 'hl'): [
+            ('cxx', 'fortran', 'hl', 'java'): [
                 'libhdf5hl_fortran',
```

Comment:
Just noticed that this should now be `libhdf5_hl_fortran` based on the cmake -> autotools change; same with ` ('fortran', 'hl')` below on new line 235.